### PR TITLE
Climate support

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change
Adding support for climate entities.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionalit)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] Local tests pass.
- [ ] There is no commented out code in this PR.
- [ ] The code has been formatted using Black (`make black`)
- [ ] Tests have been added to verify that the new code works.

### CODE REVIEW

File: custom_components/netdaemon/climate.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_SUPPORTED_FEATURES, 0)
+        )
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement used by the platform."""
+        return TEMP_CELSIUS
```
Comment (ludeeus) [inline custom_components/netdaemon/climate.py:74]: Should this be dynamic?
i.e be able to control with the service call?
```suggestion
        if not self.entity_id:
            return 0
        return (
            self._coordinator.data[self.entity_id]
            .get(ATTR_ATTRIBUTES, {})
            .get(ATTR_TEMPERATURE_UNIT, TEMP_CELSIUS)
        )
```
`ATTR_TEMPERATURE_UNIT` would be needed if suggestion is accepted

File: custom_components/netdaemon/climate.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+        Need to be a subset of HVAC_MODES.
+        """
+        if not self.entity_id:
+            return ["off"]
+
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_HVAC_MODES, [])
```
Comment (ludeeus) [inline custom_components/netdaemon/climate.py:131]:
```suggestion
            .get(ATTR_HVAC_MODES, ["off"])
```
Maybe add "off" here to have _something_

File: custom_components/netdaemon/climate.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+        Requires SUPPORT_FAN_MODE.
+        """
+        if not self.entity_id:
+            return ["off"]
+
+        return (
+            self._coordinator.data[self.entity_id]
+            .get(ATTR_ATTRIBUTES, {})
+            .get(ATTR_FAN_MODES, [])
```
Comment (ludeeus) [inline custom_components/netdaemon/climate.py:146]:
```suggestion
            .get(ATTR_FAN_MODES, ["off"])
```
Maybe add "off" here to have something

**Review by ludeeus — APPROVED** [review]

File: custom_components/netdaemon/climate.py (hunk 1 above)
Comment (helto4real) [reply custom_components/netdaemon/climate.py:73]: Good suggestion!

File: custom_components/netdaemon/climate.py [hunk 4]
```diff
@@ -102,7 +101,7 @@ def target_temperature(self) -> float:
         return (
             self._coordinator.data[self.entity_id]
             .get(ATTR_ATTRIBUTES, {})
-            .get(ATTR_TARGET_TEMPERATURE, 0.0)
+            .get(ATTR_TEMPERATURE, 0.0)
```
Comment (FrankBakkerNl) [inline custom_components/netdaemon/climate.py:111]: This seems to contradict the comment above

File: custom_components/netdaemon/climate.py [hunk 5]
```diff
@@ -113,7 +112,7 @@ def target_humidity(self) -> int:
         return (
             self._coordinator.data[self.entity_id]
             .get(ATTR_ATTRIBUTES, {})
-            .get(ATTR_TARGET_HUMIDITY, 0)
+            .get(ATTR_HUMIDITY, 0)
         )
```
Comment (FrankBakkerNl) [inline custom_components/netdaemon/climate.py:123]: Likewise

**Pull request merged**